### ISSUE TITLE
New check: Use ternary for single assignment

### ISSUE THREAD

**Issue by nickdrozd** [PyCQA/pylint#4366]
A variable gets a value assigned conditionally:

```python
if condition():
    x = 4:
else:
    x = 5
```

I don't like blocks like this because they make the situation look
more complex than it really is. A single variable is getting a value
assigned. What that value is is conditional, but the assignment
itself is not. There's one variable and one assignment, so there
should be just one statement with a ternary expression:

```python
x = 4 if condition() else 5
```

This is a rather opinionated check, so it should be added as an
extension rather than enabled by default.

**Comment by Pierre-Sassoulas** [comment]
This is so opinionated that it should have the opposite check for those who prefer not having ternary 😄

**Comment by nickdrozd** [comment]
That's a good point. Thinking about https://github.com/PyCQA/pylint/issues/4367 a little more, a streamlined way to handle these things would be to have a syntax blocklist. Then a list of forbidden keywords can be added as a `pylintrc` option. There isn't a keyword for ternaries, so that could be a boolean.

This check would still be implemented separately, because it's going the opposite way and directing the use of a certain piece of syntax.
